core-web/libs/ui/src/lib/components/dot-content-thumbnail/dot-content-thumbnail.component.ts

```diff
@@ -0,0 +1,129 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+
+const ICON_MAP = {
+    html: 'pi-file',
+    pdf: 'pi-file-pdf',
+    image: 'pi-image',
+    video: 'pi-video',
+    msword: 'pi-file-word',
+    doc: 'pi-file-word',
+    docx: 'pi-file-word'
+};
+
+export enum CONTENT_THUMBNAIL_TYPE {
+    image = 'image',
+    video = 'video',
+    icon = 'icon'
+}
+
+@Component({
+    selector: 'dot-content-thumbnail',
+    standalone: true,
+    imports: [CommonModule],
+    templateUrl: './dot-content-thumbnail.component.html',
+    styleUrls: ['./dot-content-thumbnail.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotContentThumbnailComponent implements OnInit {
+    src: string;
+    thumbnailIcon: string;
+    thumbnailType: CONTENT_THUMBNAIL_TYPE;
+
+    @Input() tempUrl: string;
+    @Input() inode: string;
+    @Input() name: string;
+    @Input() contentType: string;
+    @Input() iconSize: string;
+    @Input() titleImage: string;
+
+    readonly CONTENT_THUMBNAIL_TYPE = CONTENT_THUMBNAIL_TYPE;
+    private readonly DEFAULT_ICON = 'pi-file';
+    private type = '';
+    private thumbnailUrlMap = {
+        image: this.getImageThumbnailUrl.bind(this),
+        video: this.getVideoThumbnailUrl.bind(this),
+        pdf: this.getPdfThumbnailUrl.bind(this)
+    };
+
+    ngOnInit(): void {
+        this.type = this.contentType.split('/')[0];
+        this.setSrc();
+        this.setThumbnailType();
+        this.setThumbnailIcon();
```

Comment:
can we encapsulate this 3 methods in one? Like `initThumbnail`

Reply:
Sure